proxy_collection.go

```diff
@@ -38,6 +38,29 @@ func (collection *ProxyCollection) Add(proxy *Proxy, start bool) error {
 	return nil
 }
 
+func (collection *ProxyCollection) AddOrReplace(proxy *Proxy, start bool) error {
+	collection.Lock()
+	defer collection.Unlock()
+
+	if existing, exists := collection.proxies[proxy.Name]; exists {
+		if existing.Listen == proxy.Listen && existing.Upstream == proxy.Upstream {
```

Comment:
There's a slight problem with this, since toxiproxy will rewrite `localhost` to `127.0.0.1` and these won't always match as expected. I may put out another PR later that does a proper resolve test/compare.

The logic here is that `/populate` should be safe to rerun several times without resetting all the proxies, which currently only works if the input is static IP addresses.